src/service/debuggerCommunicationService.ts

```diff
@@ -11,15 +12,16 @@ export class DebuggerCommunicationService {
         if (this.currentDebuggerServer) {
             this.currentDebuggerServer.closeConnection();
         }
+        this.previousDebuggerServerToDisconnect = this.currentDebuggerServer;
         this.currentDebuggerServer = undefined;
     }
     public getCurrentDebuggerServer() {
         return this.currentDebuggerServer;
     }
     // Only used for stop event
     public handleStopEvent() {
-        if (this.currentDebuggerServer) {
-            this.currentDebuggerServer.disconnectFromPort();
+        if (this.previousDebuggerServerToDisconnect) {
+            this.previousDebuggerServerToDisconnect.disconnectFromPort();
         }

```

Comment:
You can replace lines 23~25 with:
`this.previousDebuggerServerToDisconnect?.disconnectFromPort()`
Same for other places.